### PR TITLE
mini-yetus: collect all the changes for processing

### PR DESCRIPTION
**Pull request by shjala**
Collect all the changes in the current branch for processing, this allows us to get yetus output before committing the changes and frees us from the hassle of rebasing to fix the yetus complaints before pushing.

### CODE REVIEW

File: tools/mini-yetus.sh [hunk 1]
```diff
@@ -79,17 +79,34 @@ fi
...
+
+all_files=$( {
+    # Committed changes between branches
+    git diff --name-only "$SRC_BRANCH".."$DST_BRANCH"
+    # Unstaged changes (modified and deleted files)
+    git diff --name-only
+    # Staged changes (modified, deleted, and added files)
+    git diff --name-only --cached
+    # Untracked files (added but not staged)
+    git ls-files --others --exclude-standard
```
[the remark this answers is not in the document]
Comment (shjala) [reply tools/mini-yetus.sh:91]: OK :D

**Comment by OhmSpectator** [comment]
Interesting, I'm on your branch, testing the tool:
```
➜  eve git:(miniyetus_imp) ✗ make mini-yetus | tail -n1 | awk '{print $5}' | xargs -I dir cat dir/results-full.txt 
home/nikolay/projects/eve/pkg/nvidia/Dockerfile:30:25:hadolint: invalid flag: --keep-git-dir
home/nikolay/projects/eve/pkg/pillar/containerd/oci_test.go:19:2:revive: should not use dot imports https://revive.run/r#dot-imports
home/nikolay/projects/eve/tools/parse-pkgs.sh:15:8:shellcheck: note: Double quote to prevent globbing and word splitting. [SC2086]
home/nikolay/projects/eve/tools/parse-pkgs.sh:64:8:shellcheck: note: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'. [SC2005]
home/nikolay/projects/eve/tools/parse-pkgs.sh:64:8:shellcheck: warning: Quote this to prevent word splitting. [SC2046]
home/nikolay/projects/eve/tools/parse-pkgs.sh:74:14:shellcheck: note: Use $(...) notation instead of legacy backticks `...`. [SC2006]
home/nikolay/projects/eve/tools/parse-pkgs.sh:74:9:shellcheck: warning: Declare and assign separately to avoid masking return values. [SC2155]
home/nikolay/projects/eve/tools/parse-pkgs.sh:84:10:shellcheck: note: Double quote to prevent globbing and word splitting. [SC2086]
home/nikolay/projects/eve/tools/parse-pkgs.sh:92:16:shellcheck: note: Double quote to prevent globbing and word splitting. [SC2086]
home/nikolay/projects/eve/tools/parse-pkgs.sh:92:8:shellcheck: note: Double quote to prevent globbing and word splitting. [SC2086]
```
It still checks the files that are untracked leftovers from the previous branches I was on...

**Comment by shjala** [comment]
> It still checks the files that are untracked leftovers from the previous branches I was on...

I'm not sure why you are seeing untracked files, according to git docs : 
- `git diff` : Changes in the working tree not yet staged for the next commit.
- `git diff --cached` : Changes between the index and your last commit; what you would be committing if you run git commit without -a option.

**Comment by OhmSpectator** [comment]
I think the reason is that the branch is not rebased on the latest master...
```
➜  eve git:(miniyetus_imp) ✗ git diff --name-only master..HEAD | cat
.github/workflows/eden_gcp.yml
.hadolint.yaml
.spdxignore
.syft.yaml
CONTRIBUTING.md
Makefile
docs/VTPM.md
images/modifiers/nvidia.yq
pkg/alpine/mirrors/3.16/community
pkg/alpine/mirrors/3.16/main
pkg/nvidia/Dockerfile
pkg/nvidia/README.md
pkg/nvidia/build.yml
pkg/nvidia/cdi/jetson-orin-nano.yaml
pkg/nvidia/cdi/jetson-xavier-nx.yaml
pkg/nvidia/patches/0001-Allow-to-build-commands-for-musl-libc.patch
pkg/nvidia/scripts/nv-init.sh
pkg/nvidia/scripts/process-cdi.sh
pkg/nvidia/udev/rules.d/90-eve-nvidia.rules
pkg/pillar/containerd/oci.go
pkg/pillar/containerd/oci_test.go
pkg/pillar/hypervisor/kvm.go
pkg/pillar/hypervisor/xen.go
pkg/xen-tools/patches-4.15.0/16-imammedo_x86_acpi_use_offset_instead_of_pointer_when_using_build_header.patch
tools/mini-yetus.sh
tools/parse-pkgs.sh
tools/spdx-check.sh
```

**Comment by shjala** [comment]
let me make it a bit more user-friendly!

**Comment by OhmSpectator** [comment]
Ah, ok. It works fine if use the up-to-date version of master now... So, nevermind...

**Pull request merged**